vsintegration/src/FSharp.Editor/CodeFix/AddMissingEqualsToTypeDefinition.fs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Threading.Tasks
+
+open Microsoft.VisualStudio.FSharp.Editor.Logging
+
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+
+[<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = "AddMissingEqualsToTypeDefinition"); Shared>]
+type internal FSharpAddMissingEqualsToTypeDefinitionCodeFixProvider() =
+    inherit CodeFixProvider()
+
+    let fixableDiagnosticIds = set ["FS0010"; "FS3360"]
+
+    override _.FixableDiagnosticIds = Seq.toImmutableArray fixableDiagnosticIds
+
+    override _.RegisterCodeFixesAsync context : Task =
+        asyncMaybe {
+            let diagnostics =
+                context.Diagnostics
+                |> Seq.filter (fun x -> fixableDiagnosticIds |> Set.contains x.Id)
+                |> Seq.toImmutableArray
+
+            for diagnostic in diagnostics do
+                let message = diagnostic.GetMessage()
+                // Ensure it is the "missing '=' in type definition" error.
+                // This is a funky heuristic but it's probably correct.
+                do! Option.guard (message.Contains("'='"))
```

Comment:
I found a counter-example when I was implementing this for ionide:

```fsharp
type Person = { Name = string }
```

in this malformed type declaration, the `=` sign will be an error of code 10, and the message will be something like "Unexpected token '=' in field declaration. yada yada yada".  This message will trigger your current logic and give a poor result. You may want to mate on these two messages specifically for the missing = on the `type Person {` style error and the `type Rabbit ()\n  member _.Blah = ()` style error.